gameFunctions.js

```diff
@@ -0,0 +1,288 @@
+    });
+}
+
+const closeNewGamePoll = (res, responseUrl, poll) => {
+    res.status(200).send();
+    const roles = assignRoles(poll, ['seer', 'doc', 'mason', 'mason']);
+    request.post({
+        url: responseUrl,
+        json: true,
+        body: {
+            text: getFormattedRoles(roles) || 'You need more than 1 player...'
+        }
+    }, (error, response, rawBody) => {
+        if (error) {
+            console.log(error);
+        }
+    });
+    notifyPlayers(roles);
+    // Keeping this around for later
+    //
+    // request.get({
+    //     url: 'https://slack.com/api/conversations.list?types=private_channel',
+    //     json: true,
+    //     headers: {
+    //         Authorization: 'Bearer ' + process.env.SLACK_AUTH_TOKEN
+    //     }
+    // }, (error, response, body) => {
+    //     if (error) {
+    //         console.log(error);
+    //     } else {
+    //         const bgChannel = body.channels.find(x => x.name == 'nothing-to-see-here');
+    //         if (bgChannel) {
+    //             console.log('Found BG Channel');
+    //             if (bgChannel.is_archived) {
+    //                 console.log('BG Channel is archived, unarchiving');
+    //                 unarchiveChannel(bgChannel.id).then(() => {
+    //                     console.log('Successfully unarchived, setting up BG channel');
+    //                     setupBGChannel(roles, bgChannel.id);
+    //                 });
+    //             } else {
+    //                 setupBGChannel(roles, bgChannel.id);
+    //             }
+    //         } else {
+    //             // createChannel().then((newChannelId) => {
+    //             //     createGroup(roles.filter(x => x.role == 'bg'), newChannelId);
+    //             // })
+    //         }
+    //     }
+    // });
+}
+
+const getFormattedRoles = (players) => {
+    let roleDisplay = 'Player Roles: \n';
+    players.sort((a,b) => a.id - b.id);
+    players.forEach((player) => {
+        roleDisplay += `\t<@${player.id}>`;
+        if (player.role) {
+            roleDisplay += `\t(${player.role.toUpperCase()})`;
+        }
+        roleDisplay += `\n`;
+    });
+    return roleDisplay;
+}
+
+const unarchiveChannel = (channelId) => {
+    return new Promise((resolve, reject) => {
+        request.post({
+            url: 'https://slack.com/api/conversations.unarchive',
+            json: true,
+            headers: {
+                Authorization: 'Bearer ' + process.env.SLACK_AUTH_TOKEN
+            },
+            body: {
+                channel: channelId,
+            }
+        }, (error, response, rawBody) => {
+            console.log(rawBody);
+            if (error) {
+                console.log(error);
+                reject();
+            }
+            resolve();
+        });
+    });
+}
+
+const setupBGChannel = (players, channelId) => {
+    console.log('Removing players from BG Channel');
+    kickGroup(players, channelId).then(() => {
```

Comment:
Should this actually kick everyone, not just the players?

Reply:
yes, but i wasn't sure how to get the current residents of the channel, so started here. This is for a future feature. I got ahead of myself